client/src/QAComponents/QAList.jsx

```diff
@@ -79,66 +48,66 @@ const QAList = (props) => {
   }
 
   const initialize = (callback) => {
-    // e.preventDefault();
     axios.get('/qa/questions' + '?product_id=' + productID)
       .then((data) => {
-        // console.log('questions received from server for product', data.data.results);
         callback(null, data.data.results)
       })
       .catch((err) => {
         console.error('error while getting product data from server')
       })
   }
 
-  // const addHelpfulAnswer = () => {
-  //   console.log('this question was helpful');
-  // }
-
-  // const reportAnswer = () => {
-  //   console.log('Answer Reported');
-  // }
+  const handleSearch = (e) => {
+    e.preventDefault()
+    setQuestions([])
+    const query = e.target.value.toLowerCase();
+    if (e.target.value.length >= 3) {
+      questionsCache.map(question => {
+        if (question.question_body.toLowerCase().includes(e.target.value)) {
+          setQuestions(prev => [...prev, question])
+        }
+      })
+    } else {
+      if (firstRender) {
+        setQuestions(questionsCache.slice(0, 2))
+      } else {
+        setQuestions(questionsCache)
+      }
+    }
+  }
 
   useEffect(() => {
     initialize((err, data) => {
       if (err) {
         console.error(err)
       } else {
-        setQuestions(prev => {
-          return prev = data.slice(0, 2)
-          // console.log('data', );
-        })
-        setAllQuestions(prev => {
-          return prev = data
-          // console.log('data', );
-        })
+        setQuestions(data.slice(0, 2))
+        setAllQuestions(data)
+        setQuestionsCache(data)
       }
     })
-    // console.log('questions', questions)
   }, [])
 
   return (
     <>
       <h1>Questions and Answers</h1>
-      {/* <Question /> */}
-      {/* <Question firstRender={firstRender}/> */}
       <form>
-        <input id='search-bar' type='text' placeholder='HAVE A QUESTION? SEARCH FOR ANSWERS' />
+        <input id='search-bar' type='text' placeholder='HAVE A QUESTION? SEARCH FOR ANSWERS' onChange={handleSearch} onSubmit={handleSearch}/>
       </form>
-      {/* {console.log('questions?', questions)} */}
-      {questions.map(question => {
-        // console.log('questions?????', question);
+      {questions ? questions.map((question, i) => {
+        let key = question.question_id + 1
         return (
           <Question
+            key={i}
             asker={question.asker_name}
             question_body={question.question_body}
```

Comment:
You might simplify the call here to the question component by just sending it the entire question and then inside of Question Component, you can use ES6 deconstruction to still access those specific values without a long chain.